app/public/cantusdata/signals/solr_sync.py

```diff
@@ -103,7 +104,9 @@ def __init__(self, solr_server_url):
         self._deletions = set()
         self._additions = set()
 
+        self._executing_flag = False
     def execute(self):
+        self._executing_flag = True
```

Comment:
Make deep copies of _deletions and _additions so they are saved before the SynchronizationSession is overwritten in SolrSynchronizer.get_session().